clientapi/routing/membership.go

```diff
@@ -46,6 +46,11 @@ func SendMembership(
 	producer *producers.RoomserverProducer,
 ) util.JSONResponse {
 	var body threepid.MembershipRequest
+
+	if req.Body == nil {
+		body = threepid.MembershipRequest{}
+		req.Body = body
```

Comment:
You can't assign to `req.Body` unfortunately as it's read only.

Line 54 is unmarshalling `req.Body`'s contents into `body`, so if you already plan to act on a `nil` body, then there's no need to Unmarshal.

Reply:
I thought that was the case. I wasn't sure what to do with the request body. Since it's nil, it'll still return in the next block with an error and I think we wanted to avoid that. I also considered doing an if statement and not unmarshaling the request if the body was nil